src/reducers/remove.js

```diff
@@ -8,9 +8,54 @@ import { Map, List } from 'immutable'
  * @return {Immutable.Map} - updated state
  */
 export const start = (state, action) => {
+  return state.withMutations(map => {
+    // Pessimistic removal
+    if (action.isAsync) {
+      // Add uid to pending.remove
+      const pendingRemove = map.getIn(['pending', 'remove'], List())
+      map.setIn(['pending', 'remove'], pendingRemove.insert(0, action.uid))
+    }
+    // Optimistic removal
+    else {
+      // Optimistically remove the item
+      map.deleteIn(['raw', action.uid])
+
+      // Keep a reference to the index of the removed item in each instance
+      let instanceIndices = List()
+      map.get('instances').entrySeq().forEach(item => {
+        const instanceKey = item[0]
+        const instanceUids = item[1].get('data')
+        const uidIndexInInstance = instanceUids.indexOf(action.uid)
+
+        // If the uid of the item to be removed exists in this instance
+        // we save the index. This allows us to re-add the item later
+        // if the removal is unsuccessful.
```

Comment:
Comment should be updated. We're simply removing the item from each instance here.